lib/sql.js

```diff
@@ -854,7 +877,14 @@ function errorIdNotFoundForUpdate(idValue) {
 SQLConnector.prototype.replaceById = function(model, id, data, options, cb) {
   if (!isIdValuePresent(id, cb)) return;
   const where = this._buildWhereObjById(model, id, data);
-  this._replace(model, where, data, options, cb);
+  try {
+    this._replace(model, where, data, options, cb);
```

Comment:
Since `_replace` is accepting a callback, I think it's more robust to catch the error inside `_replace`. IMO, async methods (accepting a callback) are expected to report errors via the callback.

Reply:
Done